pkg/objectstate/merge/merge.go

```diff
@@ -96,3 +98,19 @@ func Labels(current, updated client.Object) client.Object {
 	}
 	return updated
 }
+
+func NodeGroupConfig(current, updated nropv1alpha1.NodeGroupConfig) nropv1alpha1.NodeGroupConfig {
+	conf := nropv1alpha1.NodeGroupConfig{}
+	current.DeepCopyInto(&conf)
+
+	if updated.PodsFingerprinting != nil {
+		conf.PodsFingerprinting = updated.PodsFingerprinting
+	}
+	if updated.InfoRefreshPeriod != nil {
+		conf.InfoRefreshPeriod = updated.InfoRefreshPeriod
+	}
+	if updated.InfoRefreshMode != nil {
+		conf.InfoRefreshMode = updated.InfoRefreshMode
+	}
```

Comment:
What is the value of `InfoRefreshPeriod` if `Evented` mode is specified? IIUC, we only need `InfoRefreshPeriod` if `Periodic` mode is specified.

Reply:
this needs clarification indeed. Thanks for pointing this out.

Reply:
I think the least surprising behaviour is to propagate fields in the merge code, and make logic about the relationship between fields where we consume the data. I'm gonna change the logic which translate the settings into DS args, but keep the merge logic as simple as possible - hence updating all the fields when != nil.